python/bindings_se3.cpp

```diff
@@ -29,19 +29,36 @@ void wrap_SE3(py::module &m)
 
   SE3.def(py::init<const Scalar, const Scalar, const Scalar,
                    const Scalar, const Scalar, const Scalar>());
-  // SE3.def(py::init<const Translation&, const Quaternion&>());
+  SE3.def(py::init([](const SE3d::Translation& pos, const Eigen::Vector4d& quat) {
+                       if(abs(quat.norm() - Scalar(1)) >= manif::Constants<Scalar>::eps_s) {
+                           throw pybind11::value_error("The quaternion is not normalized!");
+                       }
+                       return manif::SE3d(pos, quat);
+                   }),
+      py::arg("position"),
+      py::arg("quaternion"));
+
   // SE3.def(py::init<const Translation&, const Eigen::AngleAxis<Scalar>&>());
   // SE3.def(py::init<const Translation&, const manif::SO3<Scalar>&>());
   // SE3.def(py::init<igen::Transform<Scalar, 3, Eigen::Isometry>&>());
 
   SE3.def("transform", &SE3d::transform);
   // SE3.def("isometry", &SE3d::isometry);
   SE3.def("rotation", &SE3d::rotation);
-  // SE3.def("quat", &SE3d::quat);
-  SE3.def(
-    "translation",
-    static_cast<SE3d::Translation (SE3d::*)(void) const>(&SE3d::translation)
-  );
+  SE3.def_property(
```

Comment:
I'd rather prefer to see `translation` and `quat` being functions rather than member variables. Maybe not very python-ish but that's more in line with the overall API and also mirroring the C++ API.

Reply:
Done in 8af8758